bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.verisure.internal;
+
+import static org.openhab.binding.verisure.internal.VerisureBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.net.CookieStore;
+import java.net.HttpCookie;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.BytesContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.openhab.binding.verisure.internal.model.VerisureAlarms;
+import org.openhab.binding.verisure.internal.model.VerisureBroadbandConnections;
+import org.openhab.binding.verisure.internal.model.VerisureClimates;
+import org.openhab.binding.verisure.internal.model.VerisureDoorWindows;
+import org.openhab.binding.verisure.internal.model.VerisureInstallations;
+import org.openhab.binding.verisure.internal.model.VerisureInstallations.Owainstallation;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection;
+import org.openhab.binding.verisure.internal.model.VerisureMiceDetection.Mouse;
+import org.openhab.binding.verisure.internal.model.VerisureSmartLock;
+import org.openhab.binding.verisure.internal.model.VerisureSmartLocks;
+import org.openhab.binding.verisure.internal.model.VerisureSmartPlugs;
+import org.openhab.binding.verisure.internal.model.VerisureThing;
+import org.openhab.binding.verisure.internal.model.VerisureUserPresences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * This class performs the communication with Verisure My Pages.
+ *
+ * @author Jarle Hjortland - Initial contribution
+ * @author Jan Gustafsson - Re-design and support for several sites and update to new Verisure API
+ *
+ */
+@NonNullByDefault
+public class VerisureSession {
+
+    @NonNullByDefault
+    private final class VerisureInstallation {
```

Comment:
Please put nested class definitions at the end of the file.

Reply:
Fixed!